public/js/clients/firefox/events.js

```diff
@@ -59,12 +65,8 @@ function resumed(_, packet) {
 function newSource(_, packet) {
   const { source } = packet;
 
-  if (NEW_SOURCE_IGNORED_URLS.indexOf(source.url) > -1) {
-    return;
-  }
-
-  if (source.introductionType == "debugger eval") {
-    return;
+  if (isHidden(source)) {
```

Comment:
You need to make sure it doesn't have a URL already. This still won't work with eval sources that are given a URL with the `sourceURL` pragma.

In fact, if we are unequivocally adding sources now, this could just be:

```js
if (!source.url) { source.url = ... }
```

You only need to create a URL if it doesn't have one.